### DISCUSSION TITLE
Run test suits in serial fashion in vitest

### DISCUSSION THREAD

**Discussion by raviSussol** [vitest-dev/vitest#1067] [General]
How can I run test suits in serial way using vitest like we do in the jest with `--runInBand` flag? I have a `build` function which builds a test server and it has 2 life cycle hook functions `beforeAll` and `afterAll` used inside that `build` function. For instance,
```
// testServer.ts

import { beforeAll, afterAll } from 'vitest';
import fastify from '../src/fastify';

const build = () => {
  const server = fastify({ logger: true });

  // setup test lifecycle hooks
  beforeAll(async () => await server.ready());
  afterAll(async () => await server.close());
  return server;
};

export default build;
```

and this function is used in multiple testing files to use that same test server. Now the problem is with command `vitest` those multiple test suits runs in parallel of which the `beforeAll` tries to be executed all at once wherever it has been used in multiple files. So initializing of same server app in same instance all at once doesn't help to run server and thus the testing suits are quit. I have a same setup but in jest and with the `--runInBand` flag the test suits run in sequential which help to start and close the server in sequential manner leading other testing suits to restart it from scratch.

So I was wondering how I can run test suits in sequential way same like it can be done in jest but using vitest. Any help would be really appreciatated. Thanks.

**Comment by sheremet-va** [comment]
You can run tests with `--no-threads`

Edit: For Vitest v2 and above: https://github.com/vitest-dev/vitest/discussions/1067#discussioncomment-12973613

**Reply by raviSussol** [reply]
Thanks @sheremet-va looks like that flag worked. But I'm getting a slight different scenarios. I have my script in the package.json as
```
// package.json
{
  ...,
  "test": "vitest --no-threads",
  ...
}
```
and when I run `yarn test` then the test runs and at the end an error is thrown saying `error Command failed with exit code 34`
![Screen Shot 2022-04-01 at 4 28 21 PM](https://user-images.githubusercontent.com/13145522/161248490-275cd13c-20b3-4d2a-96c2-2af285fc280f.png)


But if I run the same command `npx vitest --no-threads` directly in console then it executes without any errors. Do you have an idea of this?

**Comment by OssamaSijbesma** [comment]
You could use descibe.each as well, if you want a serial run for only **one** test suite.
see: https://vitest.dev/api/#describe-each

**Comment by gladykov** [comment]
Does not work anymore https://github.com/vitest-dev/vitest/issues/4620#issuecomment-1832683849

You can use `--poolOptions.forks.singleFork` with default config
